src/Main.elm

```diff
@@ -116,6 +120,74 @@ fetchMovie movie =
         Http.send LoadMovie request
 
 
+
+-- Search Justwatch.com for movie details
+
+
+searchJustWatch : MovieDetails -> Cmd Msg
+searchJustWatch movie =
+    let
+        url =
+            JustWatch.movieSearchUrl movie.movie.title
+
+        request =
+            Http.get url JustWatch.decodeJustWatchSearch
+                |> Http.toTask
+                |> andThen (handleJustWatchSearchResults movie)
+    in
+        Task.attempt (handleJustWatchDetails movie) request
+
+
+handleJustWatchSearchResults : MovieDetails -> JustWatchSearchResults -> Task.Task Http.Error JustWatchDetails
+handleJustWatchSearchResults movie searchResults =
+    let
+        match =
+            searchResults.items
+                |> List.filter (\m -> m.title == movie.movie.title)
+                |> List.head
+    in
+        case match of
+            Nothing ->
+                -- We got a response, but it didn't have any matching items, so fail with 404
+                Http.Response "" { code = 404, message = "Not Found" } Dict.empty ""
+                    |> Http.BadStatus
+                    |> Task.fail
+
+            Just result ->
+                let
+                    url =
+                        JustWatch.movieDetailUrl result.id
+                in
+                    Http.get url JustWatch.decodeJustWatchDetails
+                        |> Http.toTask
+
+
+handleJustWatchDetails movie result =
```

Comment:
For some reason I couldn't figure out the right type annotation for this function.